### ISSUE TITLE
issues with EOBS datat

### ISSUE THREAD

**Issue by larsbuntemeyer** [euro-cordex/joint-evaluation#9]
There are some [known issues](https://surfobs.climate.copernicus.eu/userguidance/known_issues_eobs.php) with EOBS data that we have to deal with, e.g., there are some regions that have only limited observations and might have to be skipped for monthly and sesaonal means. We have some experience with it. Piniging @paindeer since you have worked a lot with EOBS data to evaluate REMO ERA5 output. I could not find any details in https://doi.org/10.5194/gmd-7-1297-2014 about that.

**Comment by larsbuntemeyer** [comment]
Here is the fraction of eobs missing values from years 1980 to 2020:
```python
import xarray as xr
from dask.distributed import Client

client = Client(dashboard_address="localhost:8787")

store = "https://ncsa.osn.xsede.org/Pangeo/pangeo-forge/pangeo-forge/EOBS-feedstock/eobs-tg-tn-tx-rr-hu-pp.zarr"
ds =  xr.open_dataset(store, engine="zarr", chunks={}).sel(time=slice("1980", "2020"))


def sum_nan(da):
    return da.isnull().sum(dim="time") / da.time.size

%time tg_nan = sum_nan(ds.tg).compute()
%time pr_nan = sum_nan(ds.pp).compute()
```
```
CPU times: user 4.18 s, sys: 671 ms, total: 4.86 s
Wall time: 27.8 s
CPU times: user 2.33 s, sys: 205 ms, total: 2.54 s
Wall time: 9.47 s
```
```python
tg_nan.plot()
```
![grafik](https://github.com/user-attachments/assets/78b4a0e7-d26a-473f-9ee4-2156177a7e6e)
```python
pr_nan.plot()
```
![grafik](https://github.com/user-attachments/assets/44fc4101-5fa8-40ca-bf6b-722ae5d86198)

There are definitely some regions to take care of when computing monthly or seasonal means.

**Comment by larsbuntemeyer** [comment]
This is the issue showing up in seasonal means:
![Image](https://github.com/user-attachments/assets/33658552-9f4b-4357-9309-b6b2e271a30b)

**Comment by JavierDiezSierra** [comment]
@larsbuntemeyer Just in case it might be helpful, for the Copernicus Atlas [1], we generated a mask for E-OBS to exclude areas with fewer number of stations or where the stations were not continuous over time. It was created manually, based on simple visual prospections...

[1] https://atlas.climate.copernicus.eu/atlas/gkOVTgS5

**Comment by larsbuntemeyer** [comment]
Thanks @JavierDiezSierra , that looks reasonable. I think, if we want to look at 1980-2020, we could be a little more relaxed. I would image some criteria, where the time range of interest should not contain more than 10% (or any other threshold we can define) of missing values, othewise, it's masked out.

**Comment by larsbuntemeyer** [comment]
Here is the issue showing up in the `tas` bias:
![grafik](https://github.com/user-attachments/assets/684287c7-8966-49b4-b9a2-594e15f2e347)

**Comment by jesusff** [comment]
yes, we definitely need to apply a mask. We can start with the 10% threshold. The changes are quite sharp, so the only difference when moving the threshold will be (for pr) whether we include the circle around the eastern Mediterranean or not.
![image](https://github.com/user-attachments/assets/b4ace094-dde3-41da-a434-e0be9b41e049)

**Comment by gnikulin** [comment]
The mask from the Copernicus atlas looks better (more consistent) for me, at least visually. I would only add all missing parts of the EU countries, e.g. southern Greece, Rhodes, Cyprus, Sicily with some clarifications in the paper. Northern Africa is not in the focus and can be excluded. The eastern Mediterranean is also not in the focus, although this region can be left.

By the way, what version of E-OBS is used here ?

**Comment by JavierDiezSierra** [comment]
@gnikulin If I'm right, we are using version 23.1 at 0.1 (https://catalog.leap.columbia.edu/feedstock/eobs-dataset), but version 30.e is already available and has been dowloaded in the JSC server (/mnt/CORDEX_CMIP6_tmp/aux_data/eobs). @larsbuntemeyer, why don't we use the latest verion of E-OBS?

In Kotlarski  (2014), they regridded EUR-11 to the 0.22 E-OBS grid for the period 1989-2008, but we are regridding both projections and E-OBS to the rotated reference grid for EUR-11.

This is an example of the bias for CORDEX-CMIP5 models for the period 1989-2008 on the EUR-11 mesh. It reproduces Figure 2 from Kotlarski. The results are very similar, with slightly differences.

@jesusff 

[CMIP5_eobs_tas-4.pdf](https://github.com/user-attachments/files/18573016/CMIP5_eobs_tas-4.pdf)

**Comment by larsbuntemeyer** [comment]
As far as i see, the latest eobs is not available on 0.22 grid anymore, so i think the approach to use the EUR-11 grid for comparison seems reasonable.
